product/views.py

```diff
@@ -0,0 +1,131 @@
+                    "subcategory" : {
+                        "title"       : category.name,
+                        "description" : category.description,
+                    },
+                    "item"        : self.make_product_info_list(products=Product.objects.filter(category=category)) 
+                } for category in categories]
+
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        if Goal.objects.filter(name__icontains=sort).exists():
+            products = Product.objects.filter(goal=Goal.objects.get(name__icontains=sort))
+            result   = self.make_product_info_list(products=products) 
+            
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        if sort == 'new':
+            products = Product.objects.filter(is_new=bool(sort == 'new'))
+            result   = self.make_product_info_list(products=products) 
+
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        return JsonResponse({"message": "PAGE_NOT_FOUND"}, status=404)
+
+        
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        try:
+            product  = Product.objects.get(id=product_id)
+            category_name = product.category.menu.name
+            context  = {
+                'category'           : category_name,
+                'id'                 : product.id,
+                'productImageSrc'    : product.image_set.get(is_main=False).image_url,
+                'productCardImageSrc': product.image_set.get(is_main=True).image_url,
+                'healthGoalList'     : [goal.name for goal in product.goal.all()],
+                'title'              : product.name,
+                'subTitle'           : product.sub_name,
+                'description'        : product.description,
+                'nutritionLink'      : product.nutrition_url,
+                'allergyList'        : [allergy.name for allergy in product.allergy.all()],
+                'dietaryHabitList'   : [dietary_habit.name for dietary_habit in product.dietary_habit.all()],
+                }
+
+        # is_vegan, is_vegeterian
+            vegan_level = product.vegan_level_id
+            if vegan_level == 1:
+                is_vegan      = True
+                is_vegeterian = False
+            elif vegan_level == 2:
+                is_vegan      = False
+                is_vegeterian = True
+            else:
+                is_vegan      = False
+                is_vegeterian = False
+
+            context['isVegan']        = is_vegan
+            context['isVegetarian']   = is_vegeterian
+
+        # size, price
+            product_SSPs = ProductStock.objects.filter(product=product)  # SSP: siz e, stock, price
+            if category_name == 'vitamins':
+                price      = product_SSPs.first().price   
+                is_soldout = bool(product_SSPs.first().stock == 0)
```

Comment:
```suggestion
                is_soldout = product_SSPs.first().stock == 0
```
연산자의 return은 이미 불리언 타입이기 때문에 따로 형변환을 하는 과정을 거치는 것은 같은 작업을 두번 수행하는 것입니다.
하지 않으셔도 됩니다 :)